ee/codegen/src/generators/graph-attribute.ts

```diff
@@ -336,6 +336,36 @@ export class GraphAttribute extends AstNode {
       return;
     }
 
+    // Check if this is a cycle edge (target node already exists in the graph)
+    const isCycle = this.isNodeInBranch(targetNode, mutableAst);
+    const isSourceInGraph = this.isNodeInBranch(sourceNode, mutableAst);
+
+    if (
+      sourceNode &&
+      isCycle &&
+      isSourceInGraph &&
+      mutableAst.type === "set" &&
+      this.isDirectCycle(sourceNode, targetNode, this.workflowContext)
```

Comment:
[nit] technically speaking, we already know that we have an edge from `sourceNode >> targetNode` at this point. so we only need to check if an edge from `targetNode >> sourceNode` exists here